src/path/parts.rs

```diff
@@ -1,10 +1,15 @@
-use percent_encoding::{percent_decode_str, percent_encode, AsciiSet, CONTROLS};
+use percent_encoding::{percent_decode, percent_encode, AsciiSet, CONTROLS};
 use std::borrow::Cow;
 
-use super::DELIMITER;
+use crate::DELIMITER_BYTE;
+use snafu::Snafu;
 
-// percent_encode's API needs this as a byte
-const DELIMITER_BYTE: u8 = DELIMITER.as_bytes()[0];
+#[derive(Debug, Snafu)]
+#[snafu(display("Invalid path segment: {}", segment))]
```

Comment:
I recommend calling this `Error` for consistency

Can we please also include the reason for why the patch part is invalid in the message ? Specifically, it looks like in this case it is because the path part included percent encoded data (aka was not properly decoded?)

Reply:
This was to allow flattening the path module into the root, but perhaps better not to do this. Will revert